File: LeetCode/LeetCode Practice Problems/45- Jump Game II.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int n = nums.size();
        vector<int> a(n,INT_MAX);
        
        a[0] = 0;
        for(int i=0;i<n;i++) {
            for(int j=i+1;j<=min(n-1,i+nums[i]);j++) {
                a[j] = min(a[j] , a[i] + 1);
            }
        }
        
        
        return a[n-1];
        
    }
};

File: LeetCode/LeetCode Practice Problems/45-Jump Game II.java
class Solution {
    public int jump(int[] nums) {
        int jumps = 0, nextJump = 0, prevJump = 0;
        for(int i=0;i<nums.length;i++) {
            nextJump = Math.max(nextJump , nums[i] + i);
            if(i == prevJump && i!=nums.length-1) {
                System.out.println(i);
                prevJump = nextJump;
                jumps++;
            }
        }
        
        return jumps;
    }
}
